### PR TITLE
Adds a SlackHandler

### PR DESCRIPTION
**Pull request by terwey**

### CODE REVIEW

**Review by r4ndsen — APPROVED** [review]

File: lib/Handler/SlackHandler.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+     * channel	string	The default channel that messages will be sent to
+     * username	string	The default username for your bot
+     * icon	string	The default icon that messages will be sent with, either :emoji: or a URL to an image
+     * link_names	bool	Whether names like @regan or #accounting should be linked in the message (defaults to false)
+     * unfurl_links	bool	Whether Slack should unfurl text-based URLs (defaults to false)
+     * unfurl_media	bool	Whether Slack should unfurl media-based URLs, like tweets or Youtube videos (defaults to true)
+     * allow_markdown	bool	Whether markdown should be parsed in messages, or left as plain text (defaults to true)
+     * markdown_in_attachments	array	Which attachment fields should have markdown parsed (defaults to none)
+     */
+    public function __construct($minimalLevel, $verbose, $additionalParameters = array())
```
[the remark this answers is not in the document]
Comment (terwey) [reply lib/Handler/SlackHandler.php]: Project code consistency, it's the same everywhere else.

File: lib/Handler/SlackHandler.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        $settings = [
+            'username' => 'Flexilog',
+            'icon' => ':memo:'
+        ];
+        if (isset($additionalParameters['settings'])) {
+            $settings = array_replace($settings, $additionalParameters['settings']);
+            unset($additionalParameters['settings']);
+        }
+        parent::__construct($minimalLevel, $verbose);
+        $this->slack = new Slack\Client(
```
[the remark this answers is not in the document]
Comment (terwey) [reply lib/Handler/SlackHandler.php]: Handled by `GuzzleHttp\Exception\ConnectException`, and it should break when invalid settings are there.

File: lib/Handler/SlackHandler.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+            $settings
+        );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function write($level, $message, array $splitFullMessage, array $context = array())
+    {
+        $color = '#808080';
```
[the remark this answers is not in the document]
Comment (terwey) [reply lib/Handler/SlackHandler.php]: If you have a proposal how to do this cleaner, I'm open for ideas. Moving it to a function would call it just as often.

File: lib/Handler/SlackHandler.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (terwey) [reply lib/Handler/SlackHandler.php]: After extensive testing it doesn't break on `new Slack\Client()`, it doesn't try anything until you do a `send()` on this object.

**Pull request merged**